### ISSUE TITLE
Dead retiring worker never goes away, fills the ATC's log instead

### ISSUE THREAD

**Issue by odormond** [concourse/concourse#6990]
## Summary

When a worker with active task takes a very long time to retire and eventually completely disappear before it could complete its retirement, the ATC does not detect it's gone and keeps retrying to connect to it. While doing so, it fills the logs (around one thousand lines per second) with this kind of message:
```
{"timestamp":"2021-05-06T06:36:34.008836272Z","level":"info","source":"atc","message":"atc.tracker.notify.run.task-step.running-worker.garden-connection.retry-hijackable-client.retrying","data":{"build":"378","build_id":2339020,"error":"dial tcp 10.43.2.178:44955: connect: connection refused","failed-attempts":748240,"job":"daily-calib-ubuntu18","job-id":144777,"pipeline":"pix4d-calibration-master","ran-for":"20m20.276699529s","session":"23.26890.8.24.33.1.2","step-name":"daily-calib-ubuntu-18.04","team":"developers"}} 
```

Pruning the worker stops this madness.

## Steps to reproduce

Start a job on a worker, tell it to retire, unplug it.

## Expected results

Connection retry is throttled, ideally with exponential backoff, and eventually the worker is declared dead and buried and pruned automatically.

## Actual results

- The ATC fill up its disk.
- The performances are degraded . With multiple jobs and workers affected at the same time they are really badly affected. In our case, we are not only writing logs on disk but also sending them over the network which doesn't help.
- It might eventually have died... we did not let it go that far.

## Triaging info

* Concourse version: 7.2.0
* Runtime: garden
* Did this used to work? yes

**Comment by vito** [comment]
Hmm, I think we fixed this in https://github.com/concourse/retryhttp/pull/12 (issue: https://github.com/concourse/concourse/issues/6770) - we bumped the backoff package and the way we were constructing it prevented it from ever giving up.

But it looks like this was never tagged as a new version so it never made it home to the Concourse repo. I just tagged it with v1.1.1, so hopefully dependabot will submit a bump PR soon. I'll put this in the next milestone to make sure we pick it up. Thanks!
